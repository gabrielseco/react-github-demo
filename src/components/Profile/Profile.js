// @flow
import React from 'react';
import styles from './Profile.scss';
import type { GithubUser } from './../../types';

type Props = {
  data: GithubUser
};

const Profile = ({ data }: Props) => (
  <div className={styles.container}>
    <a href={data.html_url} target="_blank">
      <img
        className={styles.picture}
        src={data.avatar_url}
        alt={data.name}
        title={data.name}
      />
      <p className={styles.title}>{data.name}</p>
      {data.bio ? (
        <p className={styles.description}>{data.bio}</p>
      ) : (
        <p className={styles.description}>No description provided</p>
      )}
    </a>
  </div>
);

export default Profile;
