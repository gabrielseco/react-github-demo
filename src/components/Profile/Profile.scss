$font-family: 'Open Sans', sans-serif;
$border-profile-color: #cccccc;
$picture-profile-width: 150px;
$profile-color: #000000;

.container {
  border: 1px solid $border-profile-color;
  font-family: $font-family;
  overflow: auto;
  padding: 20px;
}

.picture {
  float: left;
  margin-right: 10px;
  width: $picture-profile-width;
}

.title {
  color: $profile-color;
  display: block;
  font-size: 2rem;
  margin-bottom: 15px;
}

.description {
  color: $profile-color;
  font-size: 1.2rem;
}
