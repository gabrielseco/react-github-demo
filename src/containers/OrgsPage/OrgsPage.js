// @flow
import React, { Component, Fragment } from 'react';
import debounce from 'lodash.debounce';
import GithubApi from './../../services/GithubApi';
import { Spinner, Profile, Orgs } from './../../components';
import styles from './OrgsPage.scss';
import { GithubUser, GithubOrg } from './../../types';

type State = {
  user: GithubUser,
  orgs: GithubOrg[],
  isLoading: boolean
};

type Props = {
  username: string,
  isLoading: boolean
};

class OrgsPage extends Component<Props, State> {
  constructor(props) {
    super(props);
    this.state = {
      user: undefined,
      orgs: undefined,
      isLoading: this.props.isLoading
    };

    this.searchOrgs = debounce(this.searchOrgs, 500);
  }

  componentDidMount() {
    this.update({
      isLoading: this.props.isLoading,
      username: this.props.username
    });
  }

  componentWillReceiveProps(props) {
    this.update({
      isLoading: props.isLoading,
      username: props.username
    });
  }

  update({ isLoading, username }: { isLoading: boolean, username: string }) {
    this.setState(prevState => {
      return {
        ...prevState,
        isLoading: isLoading
      };
    });

    if (username.trim() !== '') {
      this.searchOrgs(username.trim());
    } else {
      this.setState(prevState => {
        return {
          ...prevState,
          orgs: undefined,
          user: undefined,
          isLoading: false
        };
      });
    }
  }

  async searchOrgs(username: string) {
    if (username !== '') {
      const { user, orgs } = await GithubApi.getUserData(username);

      this.setState(prevState => {
        return {
          ...prevState,
          user: user,
          orgs: orgs,
          isLoading: false
        };
      });
    }
  }

  renderResults(user, orgs) {
    if (user === undefined && orgs === undefined) {
      return null;
    }

    return (
      <div className={styles.container}>
        <Profile data={user} />
        <Orgs orgs={orgs} />
      </div>
    );
  }

  render() {
    const { user, orgs, isLoading } = this.state;
    return (
      <Fragment>
        {isLoading ? (
          <div className={styles.spinnerContainer}>
            <Spinner />
          </div>
        ) : (
          this.renderResults(user, orgs)
        )}
      </Fragment>
    );
  }
}

export default OrgsPage;
