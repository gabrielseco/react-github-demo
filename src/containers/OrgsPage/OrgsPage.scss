.spinnerContainer {
  display: flex;
  justify-content: center;
  margin-top: 50px;
}

.container {
  margin-top: 10px;
  padding: 20px;
}
