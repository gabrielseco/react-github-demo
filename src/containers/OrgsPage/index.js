export { default as OrgsPage } from './OrgsPage';
