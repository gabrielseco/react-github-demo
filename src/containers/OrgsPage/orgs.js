/* eslint-disable */
export const user = {
  login: 'SaraVieira',
  id: 1051509,
  avatar_url: 'https://avatars0.githubusercontent.com/u/1051509?v=4',
  gravatar_id: '',
  url: 'https://api.github.com/users/SaraVieira',
  html_url: 'https://github.com/SaraVieira',
  followers_url: 'https://api.github.com/users/SaraVieira/followers',
  following_url:
    'https://api.github.com/users/SaraVieira/following{/other_user}',
  gists_url: 'https://api.github.com/users/SaraVieira/gists{/gist_id}',
  starred_url: 'https://api.github.com/users/SaraVieira/starred{/owner}{/repo}',
  subscriptions_url: 'https://api.github.com/users/SaraVieira/subscriptions',
  organizations_url: 'https://api.github.com/users/SaraVieira/orgs',
  repos_url: 'https://api.github.com/users/SaraVieira/repos',
  events_url: 'https://api.github.com/users/SaraVieira/events{/privacy}',
  received_events_url:
    'https://api.github.com/users/SaraVieira/received_events',
  type: 'User',
  site_admin: false,
  name: 'Sara Vieira',
  company: '@yldio ',
  blog: 'http://iamsaravieira.com',
  location: 'Portugal',
  email: null,
  hireable: null,
  bio: 'I make useless stuff for the funsies.\r\nWork at @yldio ',
  public_repos: 122,
  public_gists: 32,
  followers: 386,
  following: 34,
  created_at: '2011-09-14T21:43:39Z',
  updated_at: '2018-03-24T17:23:53Z'
};
export const orgs = [
  {
    login: 'esfiddle',
    id: 25109279,
    url: 'https://api.github.com/orgs/esfiddle',
    repos_url: 'https://api.github.com/orgs/esfiddle/repos',
    events_url: 'https://api.github.com/orgs/esfiddle/events',
    hooks_url: 'https://api.github.com/orgs/esfiddle/hooks',
    issues_url: 'https://api.github.com/orgs/esfiddle/issues',
    members_url: 'https://api.github.com/orgs/esfiddle/members{/member}',
    public_members_url:
      'https://api.github.com/orgs/esfiddle/public_members{/member}',
    avatar_url: 'https://avatars0.githubusercontent.com/u/25109279?v=4',
    description: ''
  },
  {
    login: 'OpenSourceLove',
    id: 29953818,
    url: 'https://api.github.com/orgs/OpenSourceLove',
    repos_url: 'https://api.github.com/orgs/OpenSourceLove/repos',
    events_url: 'https://api.github.com/orgs/OpenSourceLove/events',
    hooks_url: 'https://api.github.com/orgs/OpenSourceLove/hooks',
    issues_url: 'https://api.github.com/orgs/OpenSourceLove/issues',
    members_url: 'https://api.github.com/orgs/OpenSourceLove/members{/member}',
    public_members_url:
      'https://api.github.com/orgs/OpenSourceLove/public_members{/member}',
    avatar_url: 'https://avatars3.githubusercontent.com/u/29953818?v=4',
    description: ''
  },
  {
    login: 'upwithxyz',
    id: 33137346,
    url: 'https://api.github.com/orgs/upwithxyz',
    repos_url: 'https://api.github.com/orgs/upwithxyz/repos',
    events_url: 'https://api.github.com/orgs/upwithxyz/events',
    hooks_url: 'https://api.github.com/orgs/upwithxyz/hooks',
    issues_url: 'https://api.github.com/orgs/upwithxyz/issues',
    members_url: 'https://api.github.com/orgs/upwithxyz/members{/member}',
    public_members_url:
      'https://api.github.com/orgs/upwithxyz/public_members{/member}',
    avatar_url: 'https://avatars0.githubusercontent.com/u/33137346?v=4',
    description:
      'Biweekly Podcast about JavaScript, Tech, and the World... I guess'
  }
];
