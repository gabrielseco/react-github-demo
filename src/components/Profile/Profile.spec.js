import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';

import Profile from './Profile';
import { user } from './../../containers/OrgsPage/orgs';

describe('Profile suite', () => {
  it('should render the Profile component', () => {
    const comp = <Profile data={user} />;
    const component = shallow(comp);
    const tree = renderer.create(comp).toJSON();
    expect(component).toBeDefined();
    expect(tree).toMatchSnapshot();
  });
});
