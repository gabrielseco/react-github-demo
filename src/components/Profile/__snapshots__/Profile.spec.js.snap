// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Profile suite should render the Profile component 1`] = `
<div
  className="container"
>
  <a
    href="https://github.com/SaraVieira"
    target="_blank"
  >
    <img
      alt="Sara Vieira"
      className="picture"
      src="https://avatars0.githubusercontent.com/u/1051509?v=4"
      title="Sara Vieira"
    />
    <p
      className="title"
    >
      Sara Vieira
    </p>
    <p
      className="description"
    >
      I make useless stuff for the funsies.
Work at @yldio 
    </p>
  </a>
</div>
`;
